### DISCUSSION TITLE
More concise way to determine if a purchasable is in a Cart

### DISCUSSION THREAD

**Discussion by Mosnar** [craftcms/commerce#2395] [Ideas]
**What are you trying to do?**
We often need to display something different on the frontend depending on whether a purchasable is in the current user's cart or not. Currently, the only way to do that is to load the cart, grab the purchasable IDs, and then check each purchasable to see if it's in the cart. This is the case for Twig, Element API, and GraphQL and it quickly becomes cumbersome.

**What's your proposed solution?**
How and where this is implemented is up for debate, but I'm imagining most simply, it could be a method on purchasables called "qtyInCart" that would return how many of this particular item is in the current cart. Alternatively, a method on the Order would also be adequate in all cases except for GraphQL (which may have caching implications anyway). 

Here a few API examples that could be useful:

`craft.commerce.carts.cart.containsPurchasable(Purchasable purchasable|int purchasable):bool`
`craft.commerce.carts.cart.containsProduct(Product product|int product):bool`
`craft.commerce.carts.cart.getLineItemsByProduct(Product product|int product):LineItem[]`

**Comment by lukeholder** [comment]
Hi @Mosnar thanks for the feature request.

Not sure we will add this in the short-term. You can filter in twig like this in the mean-time:

```twig
    Is {{ product.defaultVariant.title }} in the cart?
    {% set filteredLineItems = cart.lineItems|filter(line => line.purchasableId == product.defaultVariant.id) %}
    {% if filteredLineItems %}Yes in the cart{% endif %}
```

You can also loop over the `filteredLineItems` to output line item details.
